flocker/node/agents/ebs.py

```diff
@@ -609,7 +609,10 @@ def compute_instance_id(self):
         """
         Look up the EC2 instance ID for this node.
         """
-        return get_instance_metadata()['instance-id'].decode("ascii")
+        instance_id = get_instance_metadata().get('instance-id', None)
+        if not instance_id:
```

Comment:
To expand on that last, it would seem quite odd to me that `get_instance_metadata` would ever return a structure where `instance-id` is missing entirely.  I might expect a failure where the instance id supplied is actually inconsistent with some other part of AWS and attempting to use it in another API call produces an error like "no such instance" - which this change would not guard against.